modules/util/portal-tools-service-builder/src/main/resources/com/liferay/portal/tools/service/builder/dependencies/service_base_impl.ftl

```diff
@@ -243,6 +246,172 @@ import org.osgi.service.component.annotations.Reference;
 			return ${entity.variableName}Persistence.update(${entity.variableName});
 		}
 
+		<#if entity.hasExternalReferenceCode() && entity.hasLazyReference() && entity.hasEntityColumn("companyId") && stringUtil.equals(entity.name, "User")>
```

Comment:
also here (and due to this "number" of places I think that probably makes sense)